src/collection-browser.ts

```diff
@@ -824,13 +859,13 @@ export class CollectionBrowser
       this.scrollToPage(this.initialPageNumber);
     }
     this.initialQueryChangeHappened = true;
+
     // if the query changed as part of a window.history pop event, we don't want to
     // persist the state because it overwrites the forward history
-
     if (!this.historyPopOccurred) {
       this.persistState();
-      this.historyPopOccurred = false;
     }
+    this.historyPopOccurred = false;
```

Comment:
This one misplaced statement is probably responsible for 90% of the back button issues we've been encountering. Sigh.